src/io/platform/f3xx/f302x8/I2Cf302x8.cpp

```diff
@@ -0,0 +1,150 @@
+#include <EVT/io/pin.hpp>
+#include <EVT/io/platform/f3xx/f302x8/I2Cf302x8.hpp>
+
+namespace EVT::core::IO {
+
+    /**
+     * Get the I2C ID associated that is associated with the given pin
+     * combination. This information is pulled from the STM32F302x8
+     * documentation with easier documentation on MBed's STM32F302r8
+     * documentation.
+     *
+     * TODO: Make modifications for this to be runnable at compile time
+     * allowing for validation of the selected pins.
+     *
+     * @param sclPin The selected I2C clock pin.
+     * @return
+     */
+    static uint8_t getPortID(Pin sclPin) {
+        switch (sclPin) {
+            case Pin::PB_8:
+            case Pin::PB_6:
+            case Pin::PA_15:
+                return 1;
+            case Pin::PA_9:
+            case Pin::PF_1:
+                return 2;
+            case Pin::PA_8:
+                return 3;
+            default:
+                return 0;
+        }
+        return 0;  // Should not get here, unless bad pin given
+    }
+
+    I2Cf302x8::I2Cf302x8(Pin sclPin, Pin sdaPin) : I2C(sclPin, sdaPin) {
+            GPIO_InitTypeDef gpioInit;
+
+        uint8_t portId = getPortID(sclPin);
+        switch (portId) {
+            case 1:
+                halI2C.Instance = I2C1;
+
+                if (!(__HAL_RCC_I2C1_IS_CLK_ENABLED()))
+                    __HAL_RCC_I2C1_CLK_ENABLE();
+
+                gpioInit.Alternate = GPIO_AF4_I2C1;
+
+                break;
+            case 2:
+                halI2C.Instance = I2C2;
+
+                if (!(__HAL_RCC_I2C2_IS_CLK_ENABLED()))
+                    __HAL_RCC_I2C2_CLK_ENABLE();
+
+                gpioInit.Alternate = GPIO_AF4_I2C2;
+
+                break;
+            case 3:
+                halI2C.Instance = I2C3;
+
+                if (!(__HAL_RCC_I2C3_IS_CLK_ENABLED()))
+                    __HAL_RCC_I2C3_CLK_ENABLE();
+
+                gpioInit.Alternate = GPIO_AF2_I2C3;
+                break;
+            default:
+                break;
+        }
+
+        Pin i2cPins[2]   = {sclPin, sdaPin};
+
+        gpioInit.Pin = static_cast<uint32_t>(1
```

Comment:
Cases like this are where the indentation comment I made above will save you a bit. Also, are we enforcing an 80-char line width? I feel like in 2021 that's really something we don't have to worry about. Lines 72, 121, 127, etc. look kinda ugly to read.

Reply:
Not anymore, technically https://www.phoronix.com/scan.php?page=news_item&px=Linux-Kernel-Deprecates-80-Col

Reply:
Ah very cool, didn't realize they had lengthened the limit. Then I think it could be worth changing it in the future.